pyfr/backends/base/makoutil.py

```diff
@@ -98,6 +98,11 @@ def macro(context, name, params, externs=''):
     params = [p.strip() for p in params.split(',')]
     externs = [e.strip() for e in externs.split(',')] if externs else []
 
+    # Ensure no white space in params/extern variables
+    for p in params+externs:
+        if ' ' in p:
```

Comment:
I'd be a bit more aggressive and validate it as `if not re.match(r'[A-Za-z_]\w*$', p):` which should catch a whole range of invalid characters for variable names.